app/scripts/modules/core/delivery/service/executions.transformer.service.js

```diff
@@ -164,7 +166,17 @@ module.exports = angular.module('spinnaker.core.delivery.executionTransformer.se
 
         var currentStage = lastRunningStage || lastFailedStage || lastNotStartedStage || lastStage;
         stage.status = currentStage.status;
-        stage.endTime = currentStage.endTime;
+        // if a stage is running, ignore the endTime of the parent stage
+        if (!currentStage.endTime) {
+          delete stage.endTime;
+        }
+        let lastEndingStage = _.max(stages, 'endTime');
+        // if the current stage has an end time (i.e. it failed or completed), use the maximum end time
+        // of all the child stages as the end time for the parent - we do this because the parent might
+        // have been an initialization stage, which ends within a few milliseconds
+        if (currentStage.endTime && lastEndingStage.endTime) {
```

Comment:
Yeah, those are the ones, I believe. Also the canary, because we do such a massive transform of that stage.

I'm 100% in favor of fixing the times in Orca if possible. In the meantime, the above changes should clean up existing executions.